base_layer/core/src/consensus.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::tari_amount::MicroTari;
+
+/// This is used to control all consensus values.
+pub struct ConsensusRules {
+    /// The min height maturity a coinbase utxo must have
+    pub coinbase_lock_height: u64,
+    /// The max range proof size that is allowed to be created
+    pub max_range_proof_range: usize,
+    /// This is emission schedule initial amount, the decay and the tail emission
+    pub emission_schedule: (MicroTari, f64, MicroTari),
+}
+
+impl Default for ConsensusRules {
+    fn default() -> Self {
+        let mut rules = ConsensusRules {
+            coinbase_lock_height: 1440,
+            max_range_proof_range: 64,
+            emission_schedule: (MicroTari::from(10_000_000), 0.999, MicroTari::from(100)),
+        };
+        if cfg!(test) {
+            rules.coinbase_lock_height = 1;
+            rules.max_range_proof_range = 32;
+        }
+        rules
+    }
+}
+
+impl ConsensusRules {
+    pub fn current(&self) -> &Self {
```

Comment:
Try using the singleton pattern here

```suggestion
    pub fn current() -> &Self {
```